app/directives/proposals/proposals.directive.js

```diff
@@ -94,29 +104,24 @@ angular.module('dashboard')
             function deleteProposal(proposal) {
                 if (proposal instanceof Object) {
                     if (proposal.isNew && self.proposals instanceof Array) {
-                        for (var index = 0; index < self.proposals.length; index++) {
-                            if (angular.equals(proposal, self.proposals[index])) {
-                                self.proposals.splice(index, 1);
-                            }
-                        }
+                        removeProposal(proposal);
                     }
                     else {
-                        if (proposal instanceof Object) {
-                            AhjoProposalsSrv.delete(proposal).$promise.then(function(response) {
-                                $log.debug("dbProposals: delete then: " + JSON.stringify(response));
-                            }, function(error) {
-                                $log.error("dbProposals: delete error: " + JSON.stringify(error));
-                            }, function(notify) {
-                                $log.debug("dbProposals: delete notify: " + JSON.stringify(notify));
-                            }).finally(function() {
-                                $log.debug("dbProposals: delete finally: ");
-                            });
-                        }
-                        else {
-                            $log.error('dbProposals: deleteProposal parameter invalid');
-                        }
+                        AhjoProposalsSrv.delete(proposal).$promise.then(function(response) {
+                            $log.debug("dbProposals: delete then: " + JSON.stringify(response));
+                            removeProposal(proposal);
```

Comment:
Could this issue deletion request to backend service and get model after it completes instead of modifying the model by UI login? I found that already troublesome today when testing backend and it failed the ops but dashboard UI still updated the model...then going back&forth brought the deleted item back to model from backend.